homeassistant/components/solarman/sensor.py

```diff
@@ -0,0 +1,314 @@
+"""Sensor platform for Solarman."""
+
+from dataclasses import dataclass
+from typing import Final
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    EntityCategory,
+    UnitOfElectricCurrent,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+    UnitOfFrequency,
+    UnitOfPower,
+    UnitOfVolume,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import SolarmanDeviceUpdateCoordinator
+from .entity import SolarmanEntity
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class SolarmanSensorEntityDescription(SensorEntityDescription):
+    """Class to describe a sensor entity."""
+
+
+SENSORS: Final = (
+    SolarmanSensorEntityDescription(
+        key="voltage",
+        translation_key="voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SolarmanSensorEntityDescription(
+        key="electric_current",
+        translation_key="electric_current",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SolarmanSensorEntityDescription(
+        key="power",
+        translation_key="power",
```

Comment:
This can have the translation key removed as it will use the device class translations